### PR TITLE
Bug 1625940 - Schedule new Hardware Report ETL

### PR DESCRIPTION
**Pull request by akkomar**

### CODE REVIEW

**Comment by akkomar** [comment]
@acmiyaguchi I'm adding a new, generic Dataproc runner script here (`jobs/moz_dataproc_runner.py`) - what do you think?

File: dags/public_data_report.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        init_actions_uris=["gs://dataproc-initialization-actions/python/pip-install.sh"],
+        additional_metadata={'PIP_PACKAGES': "git+https://github.com/mozilla/firefox-public-data-report-etl.git"},
+        additional_properties={"spark:spark.jars":"gs://spark-lib/bigquery/spark-bigquery-latest.jar",
+                               "spark-env:AWS_ACCESS_KEY_ID": aws_access_key,
+                               "spark-env:AWS_SECRET_ACCESS_KEY": aws_secret_key},
+        py_args=[
+            "public_data_report",
+            "hardware_report",
+            "--date_from", "{{ ds }}",
+            "--bq_table", "moz-fx-data-shared-prod.telemetry_derived.public_data_report_hardware",
```
[the remark this answers is not in the document]
Comment (acmiyaguchi) [reply dags/public_data_report.py:57]: Ditto

File: dags/public_data_report.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            "--temporary_gcs_bucket", params.storage_bucket,
+            "--s3_bucket", "telemetry-public-analysis-2",
+            "--s3_path", "public-data-report/hardware/",
+        ],
+        idle_delete_ttl='14400',
+        num_workers=2,
```
[the remark this answers is not in the document]
Comment (akkomar) [reply dags/public_data_report.py:65]: Not a minimum, it is actually a number of machines. At first I wanted to write it as plain Python script trying to parallelize some operations, but in the end Spark turned out to be faster and easier to develop in.

File: dags/public_data_report.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (akkomar) [reply dags/public_data_report.py:59]: I think it's not worth it - this table is pretty small, this job adds a single row per week.

File: jobs/moz_dataproc_runner.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import importlib
+import sys
+
+
+"""Generic runner for PySpark jobs
```
[the remark this answers is not in the document]
Comment (akkomar) [reply jobs/moz_dataproc_runner.py:5]: Good point, I'll file a PR.

File: dags/public_data_report.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (akkomar) [reply dags/public_data_report.py:59]: Yes, I tested this locally, following [README instructions](https://github.com/mozilla/telemetry-airflow/#testing-dataproc-jobs). Surprisingly, it was not-very-hard, thanks to all the work @acmiyaguchi did here and the documentation that @fbertsch added.

File: dags/public_data_report.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    "start_date": datetime(2020, 4, 6),
+    "email": ["telemetry-alerts@mozilla.com", "firefox-hardware-report-feedback@mozilla.com", "akomar@mozilla.com",
+              "shong@mozilla.com"],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 2,
+    "retry_delay": timedelta(minutes=10),
+}
+
+dag = DAG("public_data_hardware_report", default_args=default_args, schedule_interval="0 4 * * MON")
```
[the remark this answers is not in the document]
Comment (akkomar) [reply dags/public_data_report.py:32]: That's a leftover from old DAG. I'll look into triggering this after main ping deduplication which should be more robust.

**Comment by akkomar** [comment]
> We should hop on the deployed public data in bigquery-etl and deploy this as a JSON dataset and public BQ dataset, eventually removing the need to push the data to AWS at all.

Yes, I hope we'll do that next.

File: dags/public_data_report.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fbertsch) [reply dags/public_data_report.py:71]: Huh, I suppose the service credentials _do_ have access to the prod telemetry datasets. I wasn't sure if that was the case.

[the remark this answers is not in the document]
Comment (akkomar) [reply dags/public_data_report.py:71]: Yes, service accounts from sandbox projects do have access to prod.

**Pull request merged**